kernel/sched.c

```diff
@@ -6,6 +6,16 @@ extern const struct sched sched_bitmap;
 
 static const struct sched *sched;
 
+int sched_init()
```

Comment:
Don't make a list of available schedulers here. Instead, run hooks of each scheduler.

Reply:
The way how shell command(s) is hooked and manipulated is sort of what I expect for scheduler initialization.

Reply:
Let me confirm that. In other words, you expect that initialize internal structure of particular schedule policies rather than all available ones in `sched_init()`. And we could apply the same tactic on manipulating shell cmd(s) to do that.

Reply:
Exactly. You should always decouple scheduler implementations from generic kernel parts.

Reply:
@jserv , I think that shouldn't be finished on this PR. However, I report exactly something improper on issue #43 and solve it exactly on this PR. Your proposal could be opened on another issue, isn't it? If so, I would open another about this improvement and why we need it.

Reply:
No, it is a matter of programming techniques rather than functional change request. Do concentrate on writing code without bad smell.

Reply:
Oh, I see. I'll keep in mind. Wait for my patch.